# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1459050434 -39600
#      Sun Mar 27 14:47:14 2016 +1100
# Node ID 879f297f31945a259f30f1a0032a13f42320880e
# Parent  5f91c1e89a98eae42da9028d99044fcb8383ee35
Retain at least "all" in warning list (bug #47543)

* error.cc (set_warning_option): only remove items that are not "all".

diff --git a/libinterp/corefcn/error.cc b/libinterp/corefcn/error.cc
--- a/libinterp/corefcn/error.cc
+++ b/libinterp/corefcn/error.cc
@@ -1357,17 +1357,17 @@ set_warning_option (const std::string& s
   for (octave_idx_type i = 0; i < nel; i++)
     {
       if (tid(i).string_value () == ident)
         {
           // We found it in the current list of options.  If the state
           // for "all" is same as arg1, we can simply remove the item
           // from the list.
 
-          if (state == all_state)
+          if (state == all_state && ident != "all")
             {
               for (i = i + 1; i < nel; i++)
                 {
                   tid(i-1) = tid(i);
                   tst(i-1) = tst(i);
                 }
 
               tid.resize (dim_vector (1, nel-1));
